server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java

```diff
@@ -0,0 +1,310 @@
+        TransformationFailureException {
+        String instanceType = getInstanceType(numCpus, memSize, instanceTypes);
+        if ("".equals(instanceType)) {
+            Integer newNumCpus = numCpus;
+            Integer newMemSize = memSize;
+            //the combination does not exist
+            //try to scale cpu
+            logger.debug("The combination of numCpus: " + newNumCpus + " and memSize: " + newMemSize + " does not " +
+                "exist");
+            logger.debug("Try to scale cpu");
+            for (Integer num : allNumCpus) {
+                if (num > newNumCpus && getMemByCpu(num, instanceTypes).contains(newMemSize)) {
+                    newNumCpus = num;
+                    break;
+                }
+            }
+            instanceType = getInstanceType(newNumCpus, newMemSize, instanceTypes);
+            if ("".equals(instanceType)) {
+                logger.debug("Scaling cpu failed");
+                logger.debug("Try to scale memory");
+                //try to scale mem
+                for (Integer mem : allMemSizes) {
+                    if (mem > newMemSize && getCpuByMem(mem, instanceTypes).contains(newNumCpus)) {
+                        newMemSize = mem;
+                        break;
+                    }
+                }
+                instanceType = getInstanceType(newNumCpus, newMemSize, instanceTypes);
+                if ("".equals(instanceType)) {
+                    throw new TransformationFailureException("No combination of numCpus and memSize found");
+                } else {
+                    logger.debug("Scaling memSize succeeded, memSize: " + newMemSize);
+                }
+            } else {
+                logger.debug("Scaling numCpus succeeded, numCpus: " + newNumCpus);
+            }
+        }
+        return instanceType;
+    }
+
+    private List<Integer> getMemByCpu(Integer numCpus, ImmutableList<InstanceType> instanceTypes) {
+        return instanceTypes.stream()
+            .filter(u -> u.getNumCpus().equals(numCpus))
+            .map(InstanceType::getMemSize)
+            .collect(Collectors.toList());
+    }
+
+    private List<Integer> getCpuByMem(Integer memSize, ImmutableList<InstanceType> instanceTypes) {
+        return instanceTypes.stream()
+            .filter(u -> u.getMemSize().equals(memSize))
+            .map(InstanceType::getNumCpus)
+            .collect(Collectors.toList());
+    }
+
+    private String getInstanceType(Integer numCpus, Integer memSize, ImmutableList<InstanceType> instanceTypes) {
+        Optional<InstanceType> instanceType = instanceTypes.stream()
+            .filter(u -> u.getNumCpus().equals(numCpus) && u.getMemSize().equals(memSize))
+            .findAny();
+        if (instanceType.isPresent()) {
+            return instanceType.get().getType();
+        } else {
+            return "";
+        }
+    }
+
+    private class InstanceType {
```

Comment:
Isn't this something that should be added to the cloudformation-builder?

Reply:
Maybe, but its not concerning the template directly so i think its fine here